presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -320,10 +325,76 @@ private PlanBuilder appendQuantifiedComparisonApplyNode(PlanBuilder subPlan, Qua
+        RelationPlan subqueryRelationPlan = createRelationPlan(quantifiedComparison.getSubquery());
+        PlanNode subqueryPlan = subqueryRelationPlan.getRoot();
+        Symbol outputColumn = getOnlyElement(subqueryRelationPlan.getOutputSymbols());
+        Type outputColumnType = getOnlyElement(subqueryRelationPlan.getDescriptor().getAllFields()).getType();
+        checkState(outputColumnType.isOrderable(), "Subquery result type must be orderable");
+        FunctionRegistry functionRegistry = metadata.getFunctionRegistry();
+        Symbol subValue = symbolAllocator.newSymbol(aggregationFunction.toString(), outputColumnType);
+        subqueryPlan = new AggregationNode(
+                idAllocator.getNextId(),
+                subqueryPlan,
+                ImmutableMap.of(subValue, new FunctionCall(aggregationFunction, ImmutableList.of(outputColumn.toSymbolReference()))),
+                ImmutableMap.of(subValue, functionRegistry.resolveFunction(aggregationFunction, ImmutableList.of(outputColumnType.getTypeSignature()))),
+                ImmutableMap.of(),
+                ImmutableList.of(ImmutableList.of()),
+                AggregationNode.Step.SINGLE,
+                Optional.empty(),
+                Optional.empty());
+        subqueryPlan = new EnforceSingleRowNode(idAllocator.getNextId(), subqueryPlan);
+        ComparisonExpression valueComparedToSubquery = new ComparisonExpression(quantifiedComparison.getComparisonType(), quantifiedComparison.getValue(), subValue.toSymbolReference());
+        subPlan.getTranslations().addIntermediateMapping(quantifiedComparison, valueComparedToSubquery);
+        return appendSubqueryApplyNode(subPlan, quantifiedComparison.getSubquery(), quantifiedComparison.getSubquery(), subqueryPlan, correlationAllowed);
+    }
+
+    private static QualifiedName chooseAggregationFunction(QuantifiedComparisonExpression quantifiedComparison)
+    {
+        switch (quantifiedComparison.getQuantifier()) {
+            case ALL:
+                switch (quantifiedComparison.getComparisonType()) {
+                    case LESS_THAN:
+                    case LESS_THAN_OR_EQUAL:
+                        return QualifiedName.of("min");
+                    case GREATER_THAN:
+                    case GREATER_THAN_OR_EQUAL:
+                        return QualifiedName.of("max");
+                }
+                break;
+            case ANY:
+            case SOME:
+                switch (quantifiedComparison.getComparisonType()) {
+                    case LESS_THAN:
+                    case LESS_THAN_OR_EQUAL:
+                        return QualifiedName.of("max");
+                    case GREATER_THAN:
+                    case GREATER_THAN_OR_EQUAL:
+                        return QualifiedName.of("min");
+                }
+                break;
+        }
+        throw new IllegalStateException("Unexpected quantifier: " + quantifiedComparison.getQuantifier());
```

Comment:
Also, removed the test with VALUES 10, 'abc', because it indeed is a bit unrelated.